### ISSUE TITLE
Weird predictions after conversion to NCNN

### ISSUE THREAD

**Issue by adithya1-7** [RangiLyu/nanodet#555]
After training a custom model and converting it to NCNN, I'm getting unexpected weird detections as below. There were no issues with torch or ONNX inference. Anyone able to solve it?
When I convert the given pretrained model to NCNN, it works fine. This is strange behaviour. I suspect it has something to do with code changes that happened after 2021 

![bad](https://github.com/RangiLyu/nanodet/assets/112483428/57d06f36-4d29-4089-a53d-e6946d208b76)

**Comment by fire717** [comment]
I got a similiar problem. I trained a custom model which class num=1 and export ncnn.
I can use https://github.com/RangiLyu/nanodet/blob/main/demo_ncnn/nanodet.cpp to run predict, and got a right object,but the box is always not right.
It feels like the x and y always be smaller than right label.For example,  for a input 320x320 test image, the right box is [21,11,285,314] in [x1,y1,x2,y2] order, but the predict is [56,0,255,275].

I have change the setting in nanodet.h which not help:
```
    int input_size[2] = {320,320};
    int num_class = 1;
    int reg_max = 7; // `reg_max` set in the training config. Default: 7.
    std::vector<int> strides = { 8, 16, 32};
```
